### DISCUSSION TITLE
How to save the environment without being cleaned every time the hub it is closed?

### DISCUSSION THREAD

**Discussion by hongxrsysu** [microsoft/PlanetaryComputer#292] [Q&A]
I want to perform a remote sensing object detection task on the Planetary Computer Hub and it requires installing several new packages using pip. However, every time I disconnect, I need to reinstall all these packages as the environment seems to be cleaned up on every logout.

How can I preserve my environment and prevent it from being cleaned up? It's really inconvenient to reconfigure the environment every day.

**Comment by TomAugspurger** [comment]
The Planetary Computer Hub doesn't offer the control over the software environment that you're looking for.

You might be better off using one of the other compute options listed at https://planetarycomputer.microsoft.com/docs/concepts/computing/#use-your-own-compute.
